turbopack/crates/turbopack-ecmascript-plugins/src/transform/emotion.rs

```diff
@@ -28,15 +32,44 @@ pub enum EmotionLabelKind {
     Never,
 }
 
-//[TODO]: need to support importmap, there are type mismatch between
-//next.config.js to swc's emotion options
+#[derive(
+    Clone, PartialEq, Eq, Debug, TraceRawVcs, Serialize, Deserialize, NonLocalValue, OperationValue,
+)]
+pub struct EmotionImportItemConfig {
+    pub canonical_import: EmotionItemSpecifier,
+    pub styled_base_import: Option<EmotionItemSpecifier>,
+}
+
+impl From<&EmotionImportItemConfig> for swc_emotion::ImportItemConfig {
+    fn from(value: &EmotionImportItemConfig) -> Self {
+        swc_emotion::ImportItemConfig {
+            canonical_import: From::from(&value.canonical_import),
+            styled_base_import: value.styled_base_import.as_ref().map(From::from),
+        }
+    }
+}
+
+#[derive(
+    Clone, PartialEq, Eq, Debug, TraceRawVcs, Serialize, Deserialize, NonLocalValue, OperationValue,
+)]
+pub struct EmotionItemSpecifier(pub RcStr, pub RcStr);
+
+impl From<&EmotionItemSpecifier> for swc_emotion::ItemSpecifier {
+    fn from(value: &EmotionItemSpecifier) -> Self {
+        swc_emotion::ItemSpecifier(value.0.as_str().into(), value.1.as_str().into())
+    }
+}
+
+pub type EmotionImportMapValue = IndexMap<RcStr, EmotionImportItemConfig, FxBuildHasher>;
+
 #[turbo_tasks::value(shared, operation)]
 #[derive(Default, Clone, Debug)]
 #[serde(rename_all = "camelCase")]
 pub struct EmotionTransformConfig {
     pub sourcemap: Option<bool>,
     pub label_format: Option<String>,
     pub auto_label: Option<EmotionLabelKind>,
+    pub import_map: Option<IndexMap<RcStr, EmotionImportMapValue, FxBuildHasher>>,
```

Comment:
That isn't getting set anywhere though?

Reply:
I was also curious about it, but my conclusion was that it's created from deserialization. I'll check for test cases for this one.